### PR TITLE
Support Optional token and filter in FlyteAdminClient.listExecutions

### PR DESCRIPTION
**Pull request by narape**
# Hey, I just made a Pull Request!

## Description
Support Optional token and filter in FlyteAdminClient.listExecutions

## Motivation and Context
Paginated request to listExecutions follow this format
```
token = null
do:
  response = listExecutions(token)
  token = response.token
while(token != null)
```
We need to make sure that FlyteAdminClient accepts null tokens, the ones used for the first call instead of throwing an exception

## Have you tested this? If so, how?
I have included unit tests.

## Checklist for PR author(s)

- [ ] Changes are covered by unit test
- [ ] All tests pass
- [ ] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [ ] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-flyte-client/src/test/java/com/spotify/styx/flyte/client/TestAdminService.java [hunk 1]
```diff
@@ -76,33 +94,41 @@ public void terminateExecution(final ExecutionOuterClass.ExecutionTerminateReque
...
         .build();
     responseObserver.onNext(response);
     responseObserver.onCompleted();
   }
 
+  private boolean isEvenExecId(String execName) {
+    return Integer.parseInt(execName.substring(EXEC_NAME_PREFIX.length())) % 2 == 0;
+  }
+
+  public static ExecutionOuterClass.Execution execution(String project, String domain, String execName1) {
```
Comment (honnix) [inline styx-flyte-client/src/test/java/com/spotify/styx/flyte/client/TestAdminService.java:120]: private?

File: styx-flyte-client/src/test/java/com/spotify/styx/flyte/client/TestAdminService.java [hunk 2]
```diff
@@ -76,33 +94,41 @@ public void terminateExecution(final ExecutionOuterClass.ExecutionTerminateReque
...
 
     final ExecutionOuterClass.ExecutionList response = ExecutionOuterClass.ExecutionList.newBuilder()
         .addAllExecutions(executions)
+        .setToken(newToken)
         .build();
     responseObserver.onNext(response);
     responseObserver.onCompleted();
   }
 
+  private boolean isEvenExecId(String execName) {
```
Comment (honnix) [inline styx-flyte-client/src/test/java/com/spotify/styx/flyte/client/TestAdminService.java:116]: Where is this used?

**Review by honnix — APPROVED** [review]
Minor comments. Please feel free to merge after addressing those.

File: styx-flyte-client/src/test/java/com/spotify/styx/flyte/client/TestAdminService.java (hunk 2 above)
Comment (narape) [reply styx-flyte-client/src/test/java/com/spotify/styx/flyte/client/TestAdminService.java]: It is not

**Pull request merged**